osu.Game/Users/UserActivity.cs

```diff
@@ -87,22 +89,43 @@ public Editing(IBeatmapInfo info)
                 BeatmapInfo = info;
             }
 
-            public override string Status => @"Editing a beatmap";
+            public override string GetStatus(bool hideIdentifiableInformation = false) => @"Editing a beatmap";
         }
 
-        public class Spectating : UserActivity
+        public class Watching : UserActivity
         {
-            public override string Status => @"Spectating a game";
+            private readonly ScoreInfo score;
+
+            protected string Username => score.User.Username;
+
+            public BeatmapInfo BeatmapInfo => score.BeatmapInfo;
+
+            public Watching(ScoreInfo score)
+            {
+                this.score = score;
+            }
+
+            public override string GetStatus(bool hideIdentifiableInformation = false) => hideIdentifiableInformation ? @"Watching a game" : $@"Watching {Username}";
+        }
+
+        public class Spectating : Watching
+        {
+            public override string GetStatus(bool hideIdentifiableInformation = false) => hideIdentifiableInformation ? @"Spectating a game" : $@"Spectating {Username}";
```

Comment:
I believe "a game" doesn't fit here (sounds like a multiplayer game/room). These should have different nouns anyway, as the verbs are similar in meaning:
```suggestion
            public override string GetStatus(bool hideIdentifiableInformation = false) => hideIdentifiableInformation ? @"Watching a replay" : $@"Watching {Username}'s replay";
        }

        public class Spectating : Watching
        {
            public override string GetStatus(bool hideIdentifiableInformation = false) => hideIdentifiableInformation ? @"Spectating a user" : $@"Spectating {Username}";
```

This string also needs consideration (i.e. if it should be consistent):
https://github.com/ppy/osu/blob/7bc8908ca9c026fed1d831eb6e58df7624a8d614/osu.Game/Screens/Play/SpectatorPlayer.cs#L41